flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/FlinkDistribution.java

```diff
@@ -295,7 +295,7 @@ public void appendConfiguration(Configuration config) throws IOException {
 			.map(entry -> entry.getKey() + ": " + entry.getValue())
 			.collect(Collectors.toList());
 
-		Files.write(conf.resolve("flink-conf.yaml"), configurationLines);
+		Files.write(conf.resolve("flink-conf.yaml"), configurationLines, StandardOpenOption.APPEND);
```

Comment:
not sure what this is for, `defaultConfig` contains all options that were in the file.

Maybe you were misled by the name, but whether we actually append to the file, or write out a merged set of options is an implementation detail.